state/stateChanges/writeCollector.go

```diff
@@ -136,6 +103,47 @@ func (scc *stateChangesCollector) getStateChangesForTxs() ([]StateChangesForTx,
 	return stateChangesForTxs, nil
 }
 
+// GetStateChangesForTxs will retrieve the state changes linked with the tx hash.
+func (scc *stateChangesCollector) GetStateChangesForTxs() map[string]*data.StateChanges {
+	scc.stateChangesMut.Lock()
```

Comment:
I think a RLock would be sufficient, as you are just reading the state changes not modifying

Reply:
we are actually appending to the map in case the hash is already present. Can be seen at state/stateChanges/writeCollector.go:130